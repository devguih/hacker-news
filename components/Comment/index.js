import React from 'react'
import Link from 'next/link'
import ListComments from '../ListComments'

export default ({ item }) => (
  <article className="media">
    <figure className="media-left">
      <p className="image is-64x64">
        <img src="/static/images/user-default.jpg" alt={`Default image for user @${item.user}`} />
      </p>
    </figure>
    <div className="media-content">
      <div className="content">
        <p>
          <strong>
            <Link href={`/user/${ item.user }`}>
              <a>{ `@${item.user}` }</a>
            </Link>
          </strong>
          { ` | ` }
          <span className="has-text-grey-dark">
            { item.time_ago }
          </span>
        </p>
        <div className="has-text-grey-dark comment" dangerouslySetInnerHTML={{ __html: item.content }}></div>
      </div>
      <ListComments comments={item.comments} />
    </div>
  </article>
)
