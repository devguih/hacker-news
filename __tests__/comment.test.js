/* eslint-env jest */

import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import Comment from '../components/Comment'

const commentItem = {
  "id": 1111,
  "level": 1,
  "user": "usermock",
  "time": 1519611385,
  "time_ago": "22 minutes ago",
  "content": "Comment's description.",
  "comments": []
};

describe('<Comment />', () => {
  it('renders comment content', () => {
    const wrapper = shallow(
        <Comment item={commentItem} />
    );
    expect(wrapper.find('.comment').html()).toContain(`<div class=\"has-text-grey-dark comment\">${commentItem.content}</div>`)
  });

  it('renders user name', () => {
    const wrapper = shallow(
        <Comment item={commentItem} />
    );
    const user = wrapper.find('.user')
    expect(user.text()).toBe(`@${commentItem.user}`);
  });
});

describe('With Snapshot Testing', () => {
  it('Comment shows', () => {
    const component = renderer.create(<Comment item={commentItem} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
