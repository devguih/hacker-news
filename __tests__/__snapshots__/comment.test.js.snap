// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Comment shows 1`] = `
<article
  className="level-1 media"
>
  <figure
    className="media-left"
  >
    <p
      className="image is-64x64"
    >
      <img
        alt="Default image for user @usermock"
        src="/static/images/user-default.jpg"
      />
    </p>
  </figure>
  <div
    className="media-content"
  >
    <div
      className="content"
    >
      <p>
        <strong>
          <a
            className="user"
            href="/user/usermock"
            onClick={[Function]}
          >
            @usermock
          </a>
        </strong>
         | 
        <span
          className="has-text-grey-dark time"
        >
          22 minutes ago
        </span>
      </p>
      <div
        className="has-text-grey-dark comment"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Comment's description.",
          }
        }
      />
    </div>
    <div>
      <h2>
        Nothing found
      </h2>
    </div>
  </div>
</article>
`;

exports[`With Snapshot Testing Comment shows with children 1`] = `
<article
  className="level-1 media"
>
  <figure
    className="media-left"
  >
    <p
      className="image is-64x64"
    >
      <img
        alt="Default image for user @usermock"
        src="/static/images/user-default.jpg"
      />
    </p>
  </figure>
  <div
    className="media-content"
  >
    <div
      className="content"
    >
      <p>
        <strong>
          <a
            className="user"
            href="/user/usermock"
            onClick={[Function]}
          >
            @usermock
          </a>
        </strong>
         | 
        <span
          className="has-text-grey-dark time"
        >
          22 minutes ago
        </span>
      </p>
      <div
        className="has-text-grey-dark comment"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Comment's description.",
          }
        }
      />
    </div>
    <div>
      <article
        className="level-1 media"
      >
        <figure
          className="media-left"
        >
          <p
            className="image is-64x64"
          >
            <img
              alt="Default image for user @usermock2"
              src="/static/images/user-default.jpg"
            />
          </p>
        </figure>
        <div
          className="media-content"
        >
          <div
            className="content"
          >
            <p>
              <strong>
                <a
                  className="user"
                  href="/user/usermock2"
                  onClick={[Function]}
                >
                  @usermock2
                </a>
              </strong>
               | 
              <span
                className="has-text-grey-dark time"
              >
                22 minutes ago
              </span>
            </p>
            <div
              className="has-text-grey-dark comment"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Comment's description.",
                }
              }
            />
          </div>
          <div>
            <h2>
              Nothing found
            </h2>
          </div>
        </div>
      </article>
    </div>
  </div>
</article>
`;
